src/Kestrel.Core/Http2Limits.cs

```diff
@@ -74,5 +75,25 @@ public int MaxFrameSize
                 _maxFrameSize = value;
             }
         }
+
+        /// <summary>
+        /// Indicates the size of the maximum allowed size of a header field sequence. This limit applies to both name and value sequences in their compressed and uncompressed representations.
+        /// <para>
+        /// Value must be greater than 0, defaults to 4096
+        /// </para>
+        /// </summary>
+        public int MaxHeaderFieldSize
```

Comment:
Technically `header field` refers to the name-value pair according to the RFC. But they didn't have a specific term for the name or value portion of the header field and I'd rather keep the name as simple as possible.

Reply:
This is only for request headers, right? We should probably make that clear. We should probably name this MaxRequestHeaderFieldSize.